extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/kernel/worker/WorkerStagePhase.java

```diff
@@ -70,7 +71,7 @@ public boolean canTransitionFrom(final WorkerStagePhase priorPhase)
     @Override
     public boolean canTransitionFrom(final WorkerStagePhase priorPhase)
     {
-      return priorPhase == RESULTS_READY;
+      return priorPhase.compareTo(FINISHED) < 0;
```

Comment:
should any stage be allowed to transition to finished, even if it hasn't produced any results?

Reply:
Yes, this is needed for supporting `LIMIT`. When a downstream stage applies a `LIMIT`, it's possible for the upstream to be finished if it hasn't fully generated its output. I added a comment describing this case.